src/ecs/ecs.zig

```diff
@@ -0,0 +1,283 @@
+const std = @import("std");
+
+const main = @import("main");
+
+const ZonElement = main.ZonElement;
+
+const Entity = main.server.Entity;
+
+pub const componentlist = @import("components/_components.zig");
+pub const systemList = @import("systems/_systems.zig");
+
+const SparseSet = main.utils.SparseSet;
+
+pub const Components = listToEnum(componentlist);
+pub const Systems = listToEnum(systemList);
+
+const ComponentBitset = listToBitset(componentlist);
+const SystemBitset = listToBitset(systemList);
+const ComponentIds = listToIds(componentlist);
+const ComponentSelection = listToSelection(componentlist);
+
+var ecsArena: main.heap.NeverFailingArenaAllocator = .init(main.globalAllocator);
+pub var ecsAllocator: main.heap.NeverFailingAllocator = ecsArena.allocator();
+
+var entityTypes: SparseSet(u16, u16) = undefined;
+
+pub var componentStorage: listToSparseSets(componentlist, u16) = undefined;
+var componentIdStorage: SparseSet(ComponentIds, u32) = undefined;
+
+var componentDefaultStorage: listToSparseSets(componentlist, u16) = undefined;
+var componentBitsetStorage: [main.entity.maxEntityTypeCount]ComponentBitset = undefined;
+var systemBitsetStorage: [main.entity.maxEntityTypeCount]SystemBitset = undefined;
+
+fn listToSparseSets(comptime list: type, comptime idType: type) type {
+	var outFields: [@typeInfo(list).@"struct".decls.len]std.builtin.Type.StructField = undefined;
+
+	for (@typeInfo(list).@"struct".decls, 0..) |decl, i| {
```

Comment:
Can this be replaced with
```suggestion
	for (std.meta.declarations(list), 0..) |decl, i| {
```